src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorSpanExtensions.cs

```diff
@@ -265,7 +393,43 @@ public static TensorSpan<T> Reverse<T>(TensorSpan<T> input, nint axis = -1)
         /// </summary>
         /// <param name="input"><see cref="TensorSpan{T}"/> you want to reshape.</param>
         /// <param name="lengths"><see cref="ReadOnlySpan{T}"/> with the new dimensions.</param>
-        public static TensorSpan<T> Reshape<T>(this TensorSpan<T> input, params ReadOnlySpan<nint> lengths)
+        public static TensorSpan<T> Reshape<T>(this in TensorSpan<T> input, params ReadOnlySpan<nint> lengths)
+            where T : IEquatable<T>, IEqualityOperators<T, T, bool>
+        {
+            nint[] arrLengths = lengths.ToArray();
```

Comment:
Can we avoid intermediate allocations like this? This could stack alloc for some reasonable number of lengths.